app/src/main/java/de/danoeh/antennapod/view/ChapterSeekBar.java

```diff
@@ -0,0 +1,87 @@
+package de.danoeh.antennapod.view;
+
+import android.content.Context;
+import android.graphics.Canvas;
+import android.graphics.Paint;
+import android.util.AttributeSet;
+
+import de.danoeh.antennapod.core.util.ThemeUtils;
+
+public class ChapterSeekBar extends androidx.appcompat.widget.AppCompatSeekBar {
+
+    private float dividerMargin;
+    private float[] dividerPos;
+    private final Paint dividerPaint = new Paint();
+
+    public ChapterSeekBar(Context context) {
+        super(context);
+        init(context);
+    }
+
+    public ChapterSeekBar(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        init(context);
+    }
+
+    public ChapterSeekBar(Context context, AttributeSet attrs, int defStyle) {
+        super(context, attrs, defStyle);
+        init(context);
+    }
+
+    private void init(Context context) {
+        setBackground(null); // Removes the thumb shadow
+        dividerPos = null;
+        dividerMargin = context.getResources().getDisplayMetrics().density * 1.2f;
+        dividerPaint.setColor(ThemeUtils.getColorFromAttr(getContext(), android.R.attr.windowBackground));
+    }
+
+    /**
+     * Calculates the positions of the chapter dividers in the progress bar.
+     * @param dividerPos of the chapter dividers relative to the duration of the media.
+     */
+    public void setDividerPos(final float[] dividerPos) {
+        if (dividerPos != null && dividerPos.length > 0) {
```

Comment:
Please just store them with relative numbers and do the multiplication when drawing. Otherwise, the values become invalid after screen rotation.

Reply:
Fixed